list plugin (test) fails on OS X

steps to reproduce:
```
cd build && cmake -DPLUGINS='dump;resolver;sync;error;list' .. && make
sudo make run all
```

Then `testmod_list` fails with:
```
src/plugins/list/testmod_list.c:43: error in doTest: kdbget failed
src/plugins/list/testmod_list.c:45: error in doTest: key not found
src/plugins/list/testmod_list.c:47: error in doTest: meta key not found
src/plugins/list/testmod_list.c:48: error in doTest: kdbset failed

testmod_list RESULTS: 12 test(s) done. 4 error(s).
```

All other tests are passing.

--- a/src/plugins/list/CMakeLists.txt
+++ b/src/plugins/list/CMakeLists.txt
@@ -11,7 +11,7 @@ add_plugin(list
 list(FIND PLUGINS "rename" RENAME_INDEX)
 list(FIND PLUGINS "keytometa" KEYTOMETA_INDEX)
 
-if(BUILD_TESTING AND RENAME_INDEX AND KEYTOMETA_INDEX GREATER -1)
+if(BUILD_TESTING AND RENAME_INDEX GREATER -1 AND KEYTOMETA_INDEX GREATER -1)
 	add_plugintest(list)
 else()
 	message (WARNING "rename and keytometa plugins are required for testing, test deactivated")

--- a/src/plugins/list/CMakeLists.txt
+++ b/src/plugins/list/CMakeLists.txt
@@ -1,11 +1,18 @@
 include (LibAddMacros)
 
-add_plugin(list
-	SOURCES
-		list.h
-		list.c
-	LINK_ELEKTRA
-		elektra-kdb
-)
+list(FIND PLUGINS "rename" _rename)
+list(FIND PLUGINS "keytometa" _keytometa)
 
-add_plugintest(list)
+if(_rename GREATER -1 AND _keytometa GREATER -1)
+	add_plugin(list
+		SOURCES
+			list.h
+			list.c
+		LINK_ELEKTRA
+			elektra-kdb
+	)
+	
+	add_plugintest(list)
+else()
+	remove_plugin(list "needs plugins: rename, keytometa")
+endif()

--- a/src/plugins/list/CMakeLists.txt
+++ b/src/plugins/list/CMakeLists.txt
@@ -8,11 +8,11 @@ add_plugin(list
 		elektra-kdb
 )
 
-list(FIND PLUGINS "rename" _rename)
-list(FIND PLUGINS "keytometa" _keytometa)
+list(FIND PLUGINS "rename" RENAME_INDEX)
+list(FIND PLUGINS "keytometa" KEYTOMETA_INDEX)
 
-if(_rename AND _keytometa GREATER -1)
+if(BUILD_TESTING AND RENAME_INDEX AND KEYTOMETA_INDEX GREATER -1)
 	add_plugintest(list)
 else()
-	message (STATUS "Disabling Plugin-Test of list, depends on: rename, keytometa")
+	message (WARNING "rename and keytometa plugins are required for testing, test deactivated")
 endif()

--- a/src/plugins/list/CMakeLists.txt
+++ b/src/plugins/list/CMakeLists.txt
@@ -13,4 +13,6 @@ list(FIND PLUGINS "keytometa" _keytometa)
 
 if(_rename AND _keytometa GREATER -1)
 	add_plugintest(list)
+else()
+	message (STATUS "Disabling Plugin-Test of list, depends on: rename, keytometa")
 endif()

--- a/src/plugins/list/CMakeLists.txt
+++ b/src/plugins/list/CMakeLists.txt
@@ -1,18 +1,16 @@
 include (LibAddMacros)
 
+add_plugin(list
+	SOURCES
+		list.h
+		list.c
+	LINK_ELEKTRA
+		elektra-kdb
+)
+
 list(FIND PLUGINS "rename" _rename)
 list(FIND PLUGINS "keytometa" _keytometa)
 
-if(_rename GREATER -1 AND _keytometa GREATER -1)
-	add_plugin(list
-		SOURCES
-			list.h
-			list.c
-		LINK_ELEKTRA
-			elektra-kdb
-	)
-	
+if(_rename AND _keytometa GREATER -1)
 	add_plugintest(list)
-else()
-	remove_plugin(list "needs plugins: rename, keytometa")
 endif()

